prow/github/helpers.go

```diff
@@ -105,3 +105,156 @@ func PermissionsFromTeamPermission(permission TeamPermission) RepoPermissions {
 		return RepoPermissions{}
 	}
 }
+
+func GeneralizeComment(eventType string, event any) (*GenericCommentEvent, error) {
```

Comment:
Why do you need `eventType string`? Can't you `switch` over the type of the `event`?

Reply:
Maybe use an intersection type instead of `any` and remove the ability to error.

Reply:
👍 Let me try to switch over the event type and add a type constraint.

Reply:
Okie dokie @stevekuznetsov , thank you for taking a look at this PR 😸 

The type coercion needs to happen somewhere, so why not have it happen here? If we don't do it here, hook and each external plugin that consumes this would have to do their own coercion.

I added a type constraint for the event though, so that is safer now.

PTAL and let me know. 🙇